src/Arcus.Messaging.Pumps.Abstractions/MessagePump.cs

```diff
@@ -125,6 +154,8 @@ public virtual Task StartProcessingMessagesAsync(CancellationToken cancellationT
         public virtual Task StopProcessingMessagesAsync(CancellationToken cancellationToken)
         {
             IsStarted = false;
+            CircuitState = MessagePumpCircuitState.Open;
```

Comment:
Just wondering why stopping the pump should have an effect on the CircuitState ?

Reply:
This is reusing existing functionlity. The state is automatically set whithin the lifetime methods. When the pump is stopped, than the state is always open.